src/modules/comments/comments.service.ts

```diff
@@ -42,4 +43,29 @@ export class CommentsService {
       commentedBy,
     };
   }
+
+  async updateComment(
+    commentId: string,
+    userId: string,
+    updateCommentDto: UpdateCommentDto
+  ): Promise<CommentResponseDto> {
+    const comment = await this.commentRepository.findOneBy({ id: commentId });
```

Comment:
why not just optimize your query to retrieve the comment based on its id and the associated user. that way you need not retrieve user again to check if valid.
part of performance enhancement is to keep db calls as minimal as possible.